From 624cb7b1965487a64937bc1b0f9f79382bca3fe6 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Thu, 11 Jun 2020 16:24:21 +0800
Subject: [PATCH] media: rockchip: cif: Fix clang warning

drivers/media/platform/rockchip/cif/dev.c:290:93: warning: operator '?:' has lower precedence than '|'; '|' will be evaluated first [-Wbitwise-conditional-parentheses]
                                        (dev->chip_id != CHIP_RK1808_CIF && dev->chip_id != CHIP_RV1126_CIF) | (id == pad - 1) ?
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
drivers/media/platform/rockchip/cif/dev.c:290:93: note: place parentheses around the '|' expression to silence this warning
                                        (dev->chip_id != CHIP_RK1808_CIF && dev->chip_id != CHIP_RV1126_CIF) | (id == pad - 1) ?
                                                                                                                               ^
                                        (                                                                                     )
drivers/media/platform/rockchip/cif/dev.c:290:93: note: place parentheses around the '?:' expression to evaluate it first
                                        (dev->chip_id != CHIP_RK1808_CIF && dev->chip_id != CHIP_RV1126_CIF) | (id == pad - 1) ?
                                                                                                                               ^
                                                                                                               (

Change-Id: I7ed9acf76afeed128ccfa70f94b81e143c08e327
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/media/platform/rockchip/cif/dev.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/rockchip/cif/dev.c b/drivers/media/platform/rockchip/cif/dev.c
index 3f9aa7f10105..00c46db9a978 100644
--- a/drivers/media/platform/rockchip/cif/dev.c
+++ b/drivers/media/platform/rockchip/cif/dev.c
@@ -287,8 +287,12 @@ static int rkcif_create_links(struct rkcif_device *dev)
 					source_entity = &sensor->sd->entity;
 					sink_entity = &dev->stream[id].vnode.vdev.entity;
 
-					(dev->chip_id != CHIP_RK1808_CIF && dev->chip_id != CHIP_RV1126_CIF) | (id == pad - 1) ?
-					(flags = MEDIA_LNK_FL_ENABLED) : (flags = 0);
+					if ((dev->chip_id != CHIP_RK1808_CIF &&
+					     dev->chip_id != CHIP_RV1126_CIF) |
+					    (id == pad - 1))
+						flags = MEDIA_LNK_FL_ENABLED;
+					else
+						flags = 0;
 
 					ret = media_create_pad_link(source_entity,
 								    pad,
-- 
2.35.3

